src/uptane/uptanerepository.cc

```diff
@@ -71,16 +71,38 @@ Json::Value Repository::signVersionManifest(const Json::Value& primary_version_m
   return ecu_version_signed;
 }
 
+bool comapreForConsistency(const Uptane::Target& director_target, const Uptane::Target& image_target) {
+  if (director_target.length() > image_target.length()) return false;
+  if (director_target.filename() != image_target.filename()) return false;
+  if (director_target.hashes().size() != image_target.hashes().size()) return false;
```

Comment:
@heartsucker: is this correct?  Can you describe the desired logic for comparing Images and Director targets, please?

I think the algorithm should be:

    for each (hash algorithm known and trusted by aktualizr:
        check if that hash is in both images and director targets:
            both present & match => good
            both present & no match => reject this target
            either missing => continue to the next algorithm

Reply:
For hashes, it should be this:

```python
# python

for hash_alg in director_hashes:
  if hash_alg not in image_repo_hashes:
    abort()

for hash_alg in image_repo_hashes:
  if hash_alg not in director_hashes:
    abort()
  if image_repo_hashes[hash_alg] != director_hahes[hash_alg]:
    abort()
```

Aktualizr doesn't need to know about all hashes. E.g., aktualizr knows `sha256` and `sha512` but the backend sends `sha512/224` along with the other two. This is ok.

Reply:
OK, so: 

* The director needs to copy over exactly the set of hashes that were used to sign the image repository
* Aktualizr checks that the set of hashes match exactly between images and director
* Aktualizr then picks the best hash algorithm that it trusts.

@jerrytrieu @simao does this match your beliefs as to how the server should operate?